core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -641,17 +641,17 @@ private ServiceRequestContext newEarlyRespondingRequestContext(Channel channel,
         return new DefaultServiceRequestContext(
                 serviceConfig,
                 channel, NoopMeterRegistry.get(), protocol(),
-                nextRequestId(), routingCtx, routingResult, req.exchangeType(),
+                nextRequestId(routingCtx), routingCtx, routingResult, req.exchangeType(),
                 req, sslSession, proxiedAddresses, clientAddress,
                 System.nanoTime(), SystemInfo.currentTimeMicros());
     }
 
-    private RequestId nextRequestId() {
-        final RequestId id = config.requestIdGenerator().get();
+    private RequestId nextRequestId(RoutingContext routingCtx) {
+        final RequestId id = config.requestIdGenerator().apply(routingCtx);
```

Comment:
Sorry 😅   This isn't directly related to your change but I think it is easier for users to make a mistake when implementing `requestIdGenerator` now.

Can you also `try..catch` this line in case `requestIdGenerator()` throws an exception?
I think we can use similar warning techniques as we did for `warnedNullRequestId`.